e2e-tests/src/rewards.rs

```diff
@@ -232,14 +241,72 @@ pub fn check_points<'a>(
         max_relative_difference,
     )
 }
-pub fn get_bench_members<'a>(
-    non_reserved_members: &'a [AccountId],
-    non_reserved_members_for_session: &[AccountId],
-) -> Vec<&'a AccountId> {
-    non_reserved_members
+
+pub fn get_era_for_session<C: AnyConnection>(connection: &C, session: SessionIndex) -> EraIndex {
+    let session_first_block = get_session_first_block(connection, session);
+    get_era(connection, Some(session_first_block))
+}
+
+pub fn setup_validators(
+    config: &Config,
+) -> anyhow::Result<(EraValidators<AccountId>, CommitteeSeats, SessionIndex)> {
+    let root_connection = config.create_root_connection();
+    let era = get_current_era(&root_connection);
+    // we need to wait for at least era 1 since some of the storage items are not available at era 0
+    if era == 0 {
+        wait_for_at_least_era(&root_connection, 1)?;
+    }
+
+    let members: Vec<_> = get_validators_keys(config)
         .iter()
-        .filter(|account_id| !non_reserved_members_for_session.contains(account_id))
-        .collect::<Vec<_>>()
+        .map(account_from_keypair)
+        .collect();
+    let members_size = members.len();
+    let reserved_count = std::cmp::min(members_size / 2, 2);
```

Comment:
nit: add import for std::cmp::min

Reply:
I've change slightly logic here - seats and that list are now more dependent.